### PR TITLE
fix: propogate typenames when inteface field is used

### PR DESCRIPTION
**Pull request by JivusAyrus**
When the field of an interface is used, currently it propagates only the name of the interface in ParentTypeNames.
This PR adds all the types which implement the interface also.

### CODE REVIEW

**Review by jensneuse — CHANGES_REQUESTED** [review]
Please add a test.

File: v2/pkg/engine/plan/visitor.go [hunk 1]
```diff
@@ -383,6 +383,16 @@ func (v *Visitor) resolveFieldInfo(ref, typeRef int, onTypeNames [][]byte) *reso
 		parentTypeNames = append(parentTypeNames, onTypeName)
 	}
 
+	if v.Walker.EnclosingTypeDefinition.Kind == ast.NodeKindInterfaceTypeDefinition {
```
Comment (jensneuse) [inline v2/pkg/engine/plan/visitor.go:375]: What if the enclosing type is a union?

Comment (JivusAyrus) [reply v2/pkg/engine/plan/visitor.go:375]: Havent tested, will test

Comment (JivusAyrus) [reply v2/pkg/engine/plan/visitor.go:375]: DIscussed with david, will create a new ticket and handle it

**Review by jensneuse — APPROVED** [review]

**Pull request merged**